golem.yaml

```diff
@@ -0,0 +1,86 @@
+package:
+  name: golem
+  version: "1.2.0"
+  epoch: 0
+  description: "Golem is an open source durable computing platform that makes it easy to build and deploy highly reliable distributed systems."
+  copyright:
+    - license: Apache-2.0
+
+environment:
+  contents:
+    packages:
+      - bash
+      - build-base
+      - busybox
+      - ca-certificates-bundle
+      - cmake
+      - gcc
+      - openssl-dev
+      - pkgconf
+      - protobuf-dev
+      - protoc
+      - rustup
+      - wolfi-base
+
+pipeline:
+  - uses: git-checkout
+    with:
+      repository: https://github.com/golemcloud/golem
+      tag: v${{package.version}}
+      expected-commit: 51d5cbb1b9fcca357561c32577998ef74ce30f3d
+
+  - name: Configure and build
+    runs: |
+      rustup install stable
+      rustup target add wasm32-wasip1
+      rustup target add wasm32-unknown-unknown
+
+      # Put our rustup-installed toolchains into $PATH
+      export PATH="$(dirname "$(rustup which cargo)"):$PATH"
+
+      export PATH=$PATH:$HOME/.cargo/bin
+
+      cargo install --force cargo-make
+
+      cargo make --profile ci build-release
+
+      mkdir -p ${{targets.contextdir}}/usr/bin/
+      mv ./target/release/golem-component-service ${{targets.contextdir}}/usr/bin/golem-component-service
+
+  - uses: strip
+
+subpackages:
+  - name: ${{package.name}}-component-service
+    pipeline:
+      - runs: |
+          mkdir -p ${{targets.contextdir}}/
+          mkdir -p ${{targets.contextdir}}/config
+          mkdir -p ${{targets.contextdir}}/db
+          ln -sf /usr/bin/golem-component-service ${{targets.contextdir}}/golem-component-service
+
+          install -Dm755 ./golem-component-service/config/component-service.toml ${{targets.contextdir}}/config/component-service.toml
+          cp -R ./golem-component-service/db/* ${{targets.contextdir}}/db/
+    test:
+      environment:
+        contents:
+          packages:
+            - ${{package.name}}
+      pipeline:
+        - uses: test/tw/ldd-check
+        - name: Test symlinks
+          runs: |
+            set -e
+            test -f $(readlink -f /golem-component-service)
+
+update:
+  enabled: true
+  ignore-regex-patterns:
+    - "-dev.*"
+  github:
+    identifier: golemcloud/golem
+    strip-prefix: v
+
+test:
+  pipeline:
+    # AUTOGENERATED
+    - runs: stat /usr/bin/golem-component-service
```

Comment:
Can we expand tests to cover functional usage?

Reply:
so this is not a complete golem package (yet) we just need to build one of the components of golem named golem-component-service so we will need them all to be able to provide some good functionality test so as soon as we get a request for other services of golem I'll enhance the tests for sure